### ISSUE TITLE
Chapter 4 whereIs implementation

### ISSUE THREAD

**Issue by myrddin89** [purescript-contrib/purescript-book#366]
I’m studying the book, so maybe I missed something.
I was wondering if the following can be a better solution considering that the one proposed introduces `split` and `Pattern` which were never covered. Maybe one can suggest two different solutions.

For example the following:

```purescript
whereIs :: Path -> String -> Maybe Path
whereIs path file = head $ do
  path' <- allFiles path
  file' <- ls path'
  guard $ filename file' == filename path' <> file
  pure path'
```

**Comment by milesfrain** [comment]
Yes, we should favor solutions that don't rely on unexplained concepts. I'd be happy to replace the existing solution with yours. Feel free to open a PR.

**Comment by myrddin89** [comment]
I created a pull request at #367.

**Comment by milesfrain** [comment]
Thanks. Closed by #367
